warmup/krun_results.py

```diff
@@ -35,31 +35,38 @@
+        # We do this so we can easily check for gaps (missing pexecs).
         with open(filename, 'r') as fd:
-            data_dictionary['audit']['uname'] = uname
             reader = csv.reader(fd)
-            header = reader.next()  # Skip first row, which contains column names.
-            expect_idx = [0]  # check we get in-order indices, first always 0
-            for lineno, row in enumerate(reader):
-                # First cell contains process execution number.
-                assert int(row[0]) in expect_idx, \
-                    'Found unexpected row number %s on line %d.' % (row[0], lineno + 1)
-                bench = row[1]
-                if row[2] == 'crash':
-                    data = []
-                else:
-                    data = [float(datum) for datum in row[2:]]
-                key = '%s:%s:default-%s' % (bench, vm, language)
-                if key not in data_dictionary['wallclock_times']:
-                    data_dictionary['wallclock_times'][key] = list()
-                    data_dictionary['core_cycle_counts'][key] = list()
-                    data_dictionary['aperf_counts'][key] = list()
-                    data_dictionary['mperf_counts'][key] = list()
-                data_dictionary['wallclock_times'][key].append(data)
-                data_dictionary['core_cycle_counts'][key].append(None)
-                data_dictionary['aperf_counts'][key].append(None)
-                data_dictionary['mperf_counts'][key].append(None)
-                expect_idx = [0, int(row[0]) + 1]  # expect the next index, or a 0
+            header = reader.next()  # Skip header row.
+            rows = iter(reader)
+            sorted_rows = sorted(rows, key=lambda l: (l[1], int(l[0])))
+
+        data_dictionary['audit']['uname'] = uname
+        expect_idx = [0]  # check we get in-order indices, first always 0
+        for row in sorted_rows:
+            # First cell contains process execution number.
+            assert int(row[0]) in expect_idx, \
+                'found gaps in process executions!'
+            bench = row[1]
+            if row[2] == 'crash':
+                data = []
+            else:
+                data = [float(datum) for datum in row[2:]]
+            key = '%s:%s:default-%s' % (bench, vm, language)
+            if key not in data_dictionary['wallclock_times']:
+                data_dictionary['wallclock_times'][key] = list()
+                data_dictionary['core_cycle_counts'][key] = list()
+                data_dictionary['aperf_counts'][key] = list()
+                data_dictionary['mperf_counts'][key] = list()
+            data_dictionary['wallclock_times'][key].append(data)
+            data_dictionary['core_cycle_counts'][key].append(None)
+            data_dictionary['aperf_counts'][key].append(None)
+            data_dictionary['mperf_counts'][key].append(None)
+            expect_idx = [0, int(row[0]) + 1]  # expect the next index, or a 0
```

Comment:
"expect the next index, or a 0" -> "expect the next process execution index, or a 0"